app/controllers/application_controller.rb

```diff
@@ -9,4 +11,22 @@ def configure_permitted_parameters
     # For additional in app/views/devise/registrations/edit.html.erb
     devise_parameter_sanitizer.permit(:account_update, keys: [:username, :email, :first_name, :middle_name, :last_name, :nickname, :emoji])
   end
+
+  private
+  
+  # Sets the application's time zone based on the current user's settings. This method is a part of ensuring 
+  # that the user experience is consistent and personalized across the application. By using the time zone 
+  # set in the user's profile, all time-related data will reflect the user's local time.
+  # https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+  def set_time_zone
+    Time.zone = current_user.user_setting.time_zone if current_user.user_setting&.time_zone.present?
```

Comment:
I got rid of the `user_signed_in?` entirely bc we have the default language set to English in the `application.rb`, so if it is nil for the current user, it will be set to English. The safe operator allows us to get rid of the if statement and make the code simpler, like so:

```ruby
  def set_locale
    preferred_language = current_user&.user_setting&.preferred_language
    I18n.locale = preferred_language || I18n.default_locale
  end
```

Reply:
<img width="1168" alt="Screenshot 2024-03-26 at 17 48 10" src="https://github.com/emmvs/call_your_mom/assets/90188399/e8e21d01-570f-4162-a729-3c984f92abed">

If we go like you suggested:

```ruby
  before_action :set_locale, if: :user_signed_in?

  def set_locale
    I18n.locale = current_user&.user_setting&.preferred_language
  end
```

it saves the former language (which was Swedish in my case) which you can see at the very top right lol 🇸🇪

Reply:
Ahh interesting alright!